### PR TITLE
fix: add write perms to the default data_dir

### PR DESCRIPTION
**Pull request by pront**

### CODE REVIEW

File: .github/workflows/misc.yml [hunk 1]
```diff
@@ -48,6 +48,7 @@ jobs:
       - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
       - run: bash scripts/environment/prepare.sh
       - run: echo "::add-matcher::.github/matchers/rust.json"
+      - run: sudo chmod 755 /var/lib/vector
```
Comment (jszwedko) [inline .github/workflows/misc.yml:51]: Nit, but could we add this to `scripts/environment/bootstrap-ubuntu-20.04.sh`? That would avoid us forgetting to add it to new workflows.

**Review by jszwedko — APPROVED** [review]

File: .github/workflows/misc.yml (hunk 1 above)
Comment (pront) [reply .github/workflows/misc.yml]: Neat, good idea. Done in https://github.com/vectordotdev/vector/pull/19659/commits/cad4c0b3bc953f7253902410c7bf76d4dae30152

File: scripts/environment/bootstrap-ubuntu-20.04.sh [hunk 2]
```diff
@@ -206,3 +206,5 @@ EOF
 rustc-wrapper = "$RUSTC_WRAPPER"
 EOF
 fi
+
+sudo chmod 755 /var/lib/vector
```
Comment (bruceg) [inline scripts/environment/bootstrap-ubuntu-20.04.sh:210]: I think the `sudo` is unnecessary here. We already do things like `apt-get` above without `sudo`.

**Review by jszwedko — APPROVED** [review]

**Pull request merged**